test/e2e/app-dir/dynamic-io-errors/dynamic-io-errors.platform-dynamic.test.ts

```diff
@@ -1,119 +1,154 @@
+        return
       }
 
-      expect(next.cliOutput).toContain('◐ / ')
-      const $ = await next.render$('/')
-      expect($('[data-fallback]').length).toBe(2)
-    })
+      it('should not show a collapsed redbox error', async () => {
+        const browser = await next.browser('/')
+        await assertNoErrorToast(browser)
+      })
+    } else {
+      it('should not error the build when calling Math.random() if all dynamic access is inside a Suspense boundary', async () => {
+        try {
+          await next.start()
+        } catch {
+          throw new Error('expected build not to fail for fully static project')
+        }
+
+        expect(next.cliOutput).toContain('◐ / ')
+        const $ = await next.render$('/')
+        expect($('[data-fallback]').length).toBe(2)
+      })
+    }
   })
 
   describe('Sync Dynamic - Without Fallback - Math.random()', () => {
     const { next, isNextDev, isTurbopack, skipped } = nextTestSetup({
       files: __dirname + '/fixtures/sync-random-without-fallback',
-      skipStart: true,
+      skipStart,
       skipDeployment: true,
-      buildOptions: inDebugMode ? ['--debug-prerender'] : undefined,
+      buildOptions: inPrerenderDebugMode ? ['--debug-prerender'] : undefined,
     })
 
     if (skipped) {
       return
     }
 
     if (isNextDev) {
-      it('does not run in dev', () => {})
-      return
-    }
-
-    it('should error the build if Math.random() happens before some component outside a Suspense boundary is complete', async () => {
-      try {
-        await next.start()
-      } catch {
-        // we expect the build to fail
+      if (inPrerenderDebugMode) {
+        it('prerender debug mode does not exist for `next dev`', () => {})
+        return
       }
 
-      const output = getPrerenderOutput(next.cliOutput, {
-        isMinified: !inDebugMode,
+      it('should show a collapsed redbox error', async () => {
+        const browser = await next.browser('/')
+
+        await expect(browser).toDisplayCollapsedRedbox(`
+         {
+           "description": "Route "/" used \`Math.random()\` outside of \`"use cache"\` and without explicitly calling \`await connection()\` beforehand. See more info here: https://nextjs.org/docs/messages/next-prerender-random",
+           "environmentLabel": "Server",
+           "label": "Console Error",
+           "source": "app/page.tsx (35:23) @ RandomReadingComponent
+         > 35 |   const random = Math.random()
+              |                       ^",
+           "stack": [
+             "RandomReadingComponent app/page.tsx (35:23)",
+             "JSON.parse <anonymous> (0:0)",
+             "LogSafely <anonymous> (0:0)",
+           ],
+         }
+        `)
       })
+    } else {
+      it('should error the build if Math.random() happens before some component outside a Suspense boundary is complete', async () => {
+        try {
+          await next.start()
```

Comment:
You can just build things.
```suggestion
          await next.build()
```

Reply:
Didn't you see my note?

> [!NOTE]  
> This PR is best reviewed with hidden whitespace changes.

Reply:
https://github.com/vercel/next.js/pull/80994